fdbserver/DataDistribution.actor.cpp

```diff
@@ -3119,7 +3156,8 @@ ACTOR Future<Void> serverMetricsPolling( TCServerInfo *server) {
 //Returns the KeyValueStoreType of server if it is different from self->storeType
 ACTOR Future<KeyValueStoreType> keyValueStoreTypeTracker(DDTeamCollection* self, TCServerInfo *server) {
 	state KeyValueStoreType type = wait(brokenPromiseToNever(server->lastKnownInterface.getKeyValueStoreType.getReplyWithTaskID<KeyValueStoreType>(TaskPriority::DataDistribution)));
-	if(type == self->configuration.storageServerStoreType && (self->includedDCs.empty() || std::find(self->includedDCs.begin(), self->includedDCs.end(), server->lastKnownInterface.locality.dcId()) != self->includedDCs.end()) )
+	if(type == self->configuration.storageServerStoreType && (self->includedDCs.empty() || std::find(self->includedDCs.begin(), self->includedDCs.end(), server->lastKnownInterface.locality.dcId()) != self->includedDCs.end()) 
+		&& (self->isValidLocality(self->configuration.storagePolicy, server->lastKnownInterface.locality)) )
```

Comment:
Why is this change needed?

Reply:
This marks the wrong configured SS as wrongConfigured so that DD can remove it.

Reply:
It looks like what this is doing is preventing a valid locality from exiting. Unless this is dependent on some other change, it doesn't seem like it would have any effect on incorrectly configured servers.

Reply:
Oh I see, I misunderstood. This prevents non valid localities from doing the wait.